File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        boolean isSubsequence = false;
        int[] sIndexes = new int[s.length()];
        ;
        if (s.equals("")) {
            isSubsequence = true;
            return isSubsequence;
        }
        int startIndex = 0;
        // find the indexes of the members of s in t
        for (int i = 0; i < s.length(); i++){
            
            int index = t.indexOf(s.charAt(i),startIndex);
            if (index == -1){
                return isSubsequence;
            }
            else {
                sIndexes[i] = index;
                startIndex = index + 1;
                
                System.out.println(startIndex);
            }
        }
        for (int num : sIndexes) {
            System.out.println(num);
        }
        if (sIndexes.length == 1){
            isSubsequence = true;
            return isSubsequence;
        }
        //if s[i] < s[i + 1] for all values of s, s is a subsequence
        for (int j = 0; j < sIndexes.length - 1; j++){
            
            if (!(sIndexes[j] < sIndexes[j+1])){
                isSubsequence = false;
                return isSubsequence;
            }
            else {
                isSubsequence = true;
            }

        }
        return isSubsequence;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        isSubsequence = False
        sIndexes = []
        if s == "":
            isSubsequence = True
        startIndex = 0
        for letter in s:
            
            if letter not in t:
                return isSubsequence
            else:
                try:
                    newIndex = t.index(letter,startIndex)
                except:
                    return isSubsequence
                sIndexes.append(newIndex)
                startIndex = newIndex + 1
        if len(sIndexes) == 1:
            isSubsequence = True
        
        if len(sIndexes) == 2 and sIndexes[0] < sIndexes[1]:
            isSubsequence = True
        
        for x in range(len(sIndexes) - 2):
            
            if sIndexes[x] < sIndexes[x + 1]:
                
                isSubsequence = True



        return isSubsequence
